### PR TITLE
Add support for Microsoft.WindowsDesktop.App

### PR DESCRIPTION
**Pull request by ghost**
Addresses: https://github.com/dotnet/cli/issues/11115

### CODE REVIEW

File: scripts/obtain/dotnet-install.ps1 [hunk 1]
```diff
@@ -82,7 +83,7 @@ param(
    [string]$Version="Latest",
    [string]$InstallDir="<auto>",
    [string]$Architecture="<auto>",
-   [ValidateSet("dotnet", "aspnetcore", IgnoreCase = $false)]
+   [ValidateSet("dotnet", "aspnetcore", "windowsdesktop" IgnoreCase = $false)]
```
Comment (natemcmaster) [inline scripts/obtain/dotnet-install.ps1]:
```suggestion
   [ValidateSet("dotnet", "aspnetcore", "windowsdesktop", IgnoreCase = $false)]
```

**Review by natemcmaster — APPROVED** [review]
One syntax error, but otherwise LGTM

File: scripts/obtain/dotnet-install.ps1 (hunk 1 above)
Comment (johnbeisner) [reply scripts/obtain/dotnet-install.ps1]: Yes, saw that in testing.

**Comment by johnbeisner** [comment]
The package itself has a problem. The list of directories to unpack is empty.
From the installation output:
Command = ./dotnet-install.ps1 -runtime windowsdesktop -version 3.0.0-preview6-27707-06 -verbose
```
VERBOSE: dotnet-install: Get-List-Of-Directories-And-Versions-To-Unpack-From-Dotnet-Package
VERBOSE: dotnet-install: Directories to unpack:
```

Therefore, the WindowsDesktop runtime installed to the dlotnet root instead of 'shared'

Here is a sample output for the dotnet runtime:
Command: ./dotnet-install.ps1 -runtime dotnet -version 3.0.0-preview6-27707-06 -verbose

```
VERBOSE: dotnet-install: Get-List-Of-Directories-And-Versions-To-Unpack-From-Dotnet-Package
VERBOSE: dotnet-install: Directories to unpack: host/fxr/3.0.0-preview6-27707-06/;shared/Microsoft.NETCore.App/3.0.0-preview6-27707-06/
```

**Comment by johnbeisner** [comment]
@dagood 
I believe that the WindowsDesktop runtime zip needs repairing [see above].

**Comment by johnbeisner** [comment]
Related: https://github.com/dotnet/docs/issues/11943

**Comment by dagood** [comment]
Well, the WindowsDesktop zip DN-T produced seemed to intentionally leave out the shared dir. I kept things identical when I moved it to Core-Setup (to the best of my knowledge). I'm more than happy to get rid of the difference for simpler infra in Core-Setup but I expect a reaction will be needed in Core-SDK to handle it.

**Comment by johnbeisner** [comment]
@dagood 
Core-SDK has indeed handled it: 
https://github.com/dotnet/core-sdk/blob/d4c77e8c15e49266ac50b7dd8fed136099bca164/src/redist/targets/GenerateLayout.targets#L240-L47

**Comment by johnbeisner** [comment]
I feel that this is a different paradigm - the installation scripts work better if ‘things’ are in the same place version over version. So if we decide to standardize this later, we would potentially render some of the older WindowsDesktop zips from working correctly.

**Comment by johnbeisner** [comment]
closing for now until https://github.com/dotnet/core-setup/issues/6370 is resolved.

**Comment by dagood** [comment]
From that, it seems Core-SDK only handles the original non-`shared` implementation. I'm saying the change to have a `shared` base dir seems like another breaking change that will need shepherding. Maybe I'm reading it wrong but I'd expect at least a little conditional logic if it handles both impls.

Agree the base dir should be changed rather than adding more case-by-case logic.

**Pull request closed without merging**